std-bits/base/src/main/java/org/enso/base/spi/EnsoService.java

```diff
@@ -0,0 +1,65 @@
+package org.enso.base.spi;
+
+import java.util.logging.Logger;
+import org.enso.base.polyglot.EnsoMeta;
+import org.graalvm.polyglot.PolyglotException;
+import org.graalvm.polyglot.Value;
+
+/** A base class for an Enso service backed by an Enso type. */
+public abstract class EnsoService {
+  /**
+   * A cached Value representing the loaded type object.
+   *
+   * <p>If it is set to {@code null} it means it has not been loaded yet. If it cannot be loaded it
+   * is set to a Value instance that {@code isNull}.
+   */
+  private transient Value cachedTypeObject = null;
+
+  /**
+   * Defines the path of the Enso module that defines the type associated with this SPI
+   * registration.
+   */
+  protected abstract String getModuleName();
+
+  /** Defines the name of the type associated with this SPI registration. */
+  protected abstract String getTypeName();
+
+  /**
+   * Resolved the Enso type object associated with this SPI registration and returns it as a Value.
+   *
+   * <p>It may return {@code null} if the Enso library for the associated module is not loaded.
```

Comment:
Javadoc doesn't match the implementation.

Reply:
Thanks for pointing out - indeed I forgot to update it.